### ISSUE TITLE
Patches are generated in wrong order after `keydown`

### ISSUE THREAD

**Issue by tomalec** [PuppetJs/PuppetJs#98]
It seems that after removal of `keydown` from fast-json-patch (https://github.com/Starcounter-Jack/JSON-Patch/issues/98), we introduced another bug.

## Steps to reproduce

1.  Create two elements that binds data to change and click events respectively, e.g.

   ```html
<input type="text"value="{{model.B_Value::change}}" onchange="console.log('change')" placeholder="type first">
<button value="{{model.A_Clicks::click}}" onclick="++this.value">then click here</button>
```

2. Type something into input
3. Press the button

## Expected result

As `onchange` is triggered before `onclick`, we should get operation objects in same order:
```js
{op:'replace', path: '/model/B_Value', value: 'smth'},

{op:'replace', path: '/model/A_Clicks', value: '1'}
```

## Actual result

Dirty checking is done only after the `click` event so it results in single patch with
```js
{op:'replace', path: '/model/A_Clicks', value: '1'},
{op:'replace', path: '/model/B_Value', value: 'smth'}
```

### 1. Revert the change with `keydown`/`mousedown`
So we will dirty-check between actions

### 2. Replace dom observers with deeper Polymer notification protocol integration
So we will not wait for events, but for children to parent notification that the property of the view-model was changed.

//cc @andwah, @warpech, @aloscha 

PS> @andwah - this is the reason why Warehouse Client app adds empty Store Keys

**Comment by warpech** [comment]
Aren't the patches generated in the order in which they are defined in JSON?

**Comment by warpech** [comment]
Perhaps 1) is indeed the best solution until we have us JS `Proxy` some time in future

**Comment by tomalec** [comment]
Yest, they are, but we want them to be generated in order they happened, and workaround involving renaming/redefining JSON is extremely ugly.

**Comment by warpech** [comment]
Ok, then solution 1) is agreed. @tomalec could you please make it?

**Comment by tomalec** [comment]
great, sure

**Comment by miyconst** [comment]
Thanks for resolving this, because it also hits me from time to time.

**Comment by Mnosek** [comment]
Thanks a lot for taking this! I've also faced this problem today. Would be great to have it working correctly.

**Comment by tomalec** [comment]
Fixed in [2.2.1](https://github.com/PuppetJs/PuppetJs/releases/tag/2.2.1)
(Starcounter>2.2.1.3157)
